dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ResourcesController.java

```diff
@@ -270,6 +270,25 @@ public Result<Object> deleteResource(@Parameter(hidden = true) @RequestAttribute
         return resourceService.delete(loginUser, fullName, tenantCode);
     }
 
+    /**
+     * delete DATA_TRANSFER data
+     *
+     * @param loginUser login user
+     * @return delete result code
+     */
+    @Operation(summary = "deleteDataTransferData", description = "Delete the N days ago data of DATA_TRANSFER ")
+    @Parameters({
+            @Parameter(name = "days", description = "N days ago", required = true, schema = @Schema(implementation = Integer.class))
+    })
+    @DeleteMapping(value = "/data-transfer-delete")
```

Comment:
Don't need delete action due to the request method is `DeleteMapping`.
```suggestion
    @DeleteMapping(value = "/data-transfer")
```